lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/DefaultVortexManager.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver;
+
+import org.apache.reef.vortex.driver.tasklet.RunningExecutors;
+import org.apache.reef.vortex.driver.tasklet.Tasklet;
+import org.apache.reef.vortex.driver.tasklet.TaskletExecutor;
+
+import javax.inject.Inject;
+import java.io.Serializable;
+import java.util.*;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Default implementation of VortexManager.
+ */
+public class DefaultVortexManager implements VortexManager {
+  private static final Logger LOG = Logger.getLogger(DefaultVortexManager.class.getName());
+  private final RunningExecutors runningExecutors;
+  private final BlockingDeque<Tasklet> pendingTasklets = new LinkedBlockingDeque<>();
+  private final ExecutorService schedulerThread = Executors.newSingleThreadExecutor();
```

Comment:
I personally prefer the current `ExecutorService`
* This thread directly accesses pendingTasklets/runningExecutors, which currently are directly accessed only by methods in this class. It maybe better to keep the accesses all in the same class for readability.
* I'm not sure how we can map execution of an infinite loop to the `onNext(event)` interface well.